modules/common/src/main/scala/surge/core/Controllable.scala

```diff
@@ -2,19 +2,48 @@
 
 package surge.core
 
+import scala.concurrent.Future
+
+final case class ControlAck(success: Boolean, error: Option[Throwable] = None) extends Ack {
+  override def withSuccess(success: Boolean): Ack = {
+    copy(success = success)
+  }
+
+  override def withError(error: Throwable): Ack = {
+    copy(error = Some(error))
+  }
+}
+
+trait Ack {
```

Comment:
I think we can simplify this even further and have the `Ack` just be a simple case object.  In a handful of places we look at if success is true and return a failed future if it's false, but I think we can lean more on that and have the success = false case just be a failed future rather than a successful future with success = false. That would consolidate the handling of the couple of failure cases as well (success = false and failed future)

Reply:
done